great_expectations/datasource/fluent/sql_datasource.py

```diff
@@ -830,6 +831,7 @@ class SQLDatasource(Datasource):
     # left side enforces the names on instance creation
     type: Literal["sql"] = "sql"
     connection_string: Union[ConfigStr, str]
+    create_temp_table: bool = True
```

Comment:
This isn't getting passed to `create_engine`, but it will once this PR goes in.

https://github.com/great-expectations/great_expectations/pull/7394/files#diff-423da4f606a5cb65bd03d9733b0a26bdaa75e45d94d9047cab0426c9719a2a78

Reply:
This doesn't get passed to SQL Alchemy `create_engine`. This is used by the GX `SQLAlchemyExecutionEngine.

Reply:
Where does that happen?

Reply:
Also, I spoke incorrectly earlier, this won't get passed to `create_engine` automatically. So if it shouldn't be passed, you don't need to do anything.

https://github.com/great-expectations/great_expectations/blob/a618424d50c86c18ac1877a83e2213c8a0645596/great_expectations/datasource/fluent/sql_datasource.py#L863-L869

Reply:
Because it is not listed [here](https://github.com/great-expectations/great_expectations/blob/05cd82abb41b4b6b160583c61a3cb83dedadae26/great_expectations/datasource/fluent/interfaces.py#L354) it will get passed [here](https://github.com/great-expectations/great_expectations/blob/05cd82abb41b4b6b160583c61a3cb83dedadae26/great_expectations/datasource/fluent/interfaces.py#L422). You can also see in the tests I added that the `_create_temp_table` attribute gets set on our `MockSqlAchemyExecutionEngine`.